pulp_rpm/app/models.py

```diff
@@ -760,3 +784,143 @@ class ModulemdDefaults(Content):
+    # platforms = <str>[,<str>...]          ; equal to [tree]/platforms
+    # packagedir = <str>                    ; equal to [variant-*]/packages
+    # repository = <str>                    ; equal to [variant-*]/repository
+    # timestamp = <int>                     ; equal to [tree]/build_timestamp
+    # variant = <str>                       ; variant UID of first variant (sorted alphabetically)
+
+    @property
+    def checksums(self, rehash=False):
+        """
+        Return checksums of images.
+
+        """
+        if getattr(self, '_sha256', None) is None or rehash is True:
+            self._sha256 = []
+
+            for platform in self.tree_platforms():
+                hashes = []
+
+                for image in platform.images():
+                    with (image.name, "rb") as file:
+                        sha256 = hashlib.sha256(file).hexdigest()
+                    checksum_dict = {image.path: {"sha256": sha256}}
+                    hashes.append(checksum_dict)
+
+                self._sha256.extend(hashes)
+
+        return self._sha256
+
+    class Meta:
+        default_related_name = "%(app_label)s_%(model_name)s"
+
+
+class TreePlatform(Model):
+    """
+    Distribution Tree PLatform.
+
+    """
+
+    name = models.CharField(max_length=20)
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='tree_platforms'
+    )
+
+
+class Images(Model):
+    """
+    Distribution Tree Images.
+
+    """
+
+    name = models.CharField(max_length=20)
+    path = models.CharField(max_length=50)
+    platform = models.ForeignKey(TreePlatform, on_delete=models.CASCADE, related_name='images')
+
+
+class Addons(Model):
+    """
+    Distribution Tree Addons.
+
+    """
+
+    id = models.CharField(max_length=30)
+    uid = models.CharField(max_length=30)
+    name = models.CharField(max_length=50)
+    type = models.CharField(max_length=20)
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='addons'
+    )
+    repository = models.ForeignKey(
+        Repository, on_delete=models.CASCADE, related_name='addons'
+    )
+
+
+class DistributionTreeRepository(Model):
+    """
+    Repository for "distribution_tree" content.
+
+    """
+
+    TYPE = 'distribution_tree'
+
+    id = models.CharField(max_length=30)
+    uid = models.CharField(max_length=30)
+    name = models.CharField(max_length=50)
+    type = models.CharField(max_length=20)
+    packages = models.CharField(max_length=50)
+    source_packages = models.CharField(max_length=50)
+    source_repository = models.CharField(max_length=50)
+    debug_packages = models.CharField(max_length=50)
+    debug_repository = models.CharField(max_length=50)
+    identity = models.CharField(max_length=50)
+    content = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='tree_variants'
```

Comment:
I would probably just call this related_name variants since it's a top level item.